### PR TITLE
New Component DatePickerFullScreen

### PR DESCRIPTION
**Pull request by cerinman**
@mxenabled/frontend-engineers 

This PR adds a new component.  DatePickerFullScreen.

This component is a copy of the DatePicker component but the calendar is now a full screen scrim that covers it's parent element.

screen shot 2015-11-30 at 3 41 59 pm
screen shot 2015-11-30 at 3 42 16 pm

Props:

- isFixed Type(boolean) - Default(False): Used to determine if the calendar scrim is positioned fixed or absolute.
- caretWrapperStyle Type(object): Used to style the caret icon on the selected date.
- closeIcon Type(string) - Default('close'): Name of icon used to select the close icon on the calendar scrim.
- closeOnDateSelect Type(boolean) - Default(False): Determines if the calendar scrim is closed on date select or not.
- defaultDate Type(number): Unix time stamp. The default date for the calendar and selected date.
- format Type(string) - Default('MMM D, YYYY'): The format for the selected date.
- inputStyle Type(object): Used to style the selected date input.
- locale Type(string) - Default('en'): Used for time zone offset when formatting dates.
- minimumDate Type(number): Unix time stamp. Sets the earliest date which a user can select.
- onDateSelect Type(function): A supplied call back function that fires on date select and returns the selected date.
- placeholderText Type(string): The place holder text for the selected date input field.
- placeholderTextStyle Type(object): Used to style the place holder text.
- selectedDateWrapperStyle Type(object): Used to style the selected date div wrapping element.
- showCaret Type(boolean) - Default(true): Determines if the caret icon is shown on the selected date.
- showDayBorders Type(boolean) - Default(false): Determines if the calendar dates have borders or not.
- style Type(object): Used to style component.
- title Type(string) - Default('Select A Date'): The title placed above the calendar.
- useInputForSelectedDate Type(boolean) - Default(true): Determines if the selected date is rendered as plain text or placed in an input field.

Classes:

- mx-date-picker-full-screen: Used to style component.
- mx-date-picker-full-screen-selected-date: Used to style the selected date.
- mx-date-picker-full-screen-calendar-scrim: Used to style the calendar scrim.
- mx-date-picker-full-screen-calendar-wrapper: Used to style the div wrapping the calendar.
- mx-date-picker-full-screen-calendar-header: Used to style the calendar header.

### CODE REVIEW

File: src/components/DatePickerFullScreen.js [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+            <div style={styles.clearFix}></div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+DatePickerFullScreen.propTypes = {
+  calendarScrimFixedPosition: React.PropTypes.bool,
```
Comment (bsbeeks) [inline src/components/DatePickerFullScreen.js]: I would call this `isFixed` instead.

File: src/components/DatePickerFullScreen.js [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+  onDateSelect () {},
+  showCaret: true,
+  showDayBorders: false,
+  title: 'Select A Date',
+  useInputForSelectedDate: true
+};
+
+const styles = {
+  caret: {
+    color: '#CCCCCC',
```
Comment (bsbeeks) [inline src/components/DatePickerFullScreen.js]: Could we update all the colors to be pulled from the Style constants file? Refer to frontify for the color pallet to use.

**Comment by cerinman** [comment]
@bsbeeks fixed naming issue and moved all font sizes/colors to values from the constants file.  Also removed px where ever appropriate.  

Here is is after the changes.
screen shot 2015-12-01 at 8 56 21 am
screen shot 2015-12-01 at 8 56 29 am

**Comment by bsbeeks** [comment]
One last change, we need to be able to change the close icon through a prop.

**Comment by cerinman** [comment]
@bsbeeks The prop has been added. Ready for another review.

**Comment by bsbeeks** [comment]
:shipit: make sure we update the docs once we release this.

**Pull request merged**